sympy/physics/optics/waves.py

```diff
@@ -246,6 +251,12 @@ def __str__(self):
 
     __repr__ = __str__
 
+    def __mul__(self, float):
+        return TWave(self._amplitude*float, self._frequency, self._phase)
```

Comment:
IMO the fix here (https://github.com/sympy/sympy/pull/8908) is better. You are not checking for the type of your variable `float`.